File: Presentation Layer (frontend)/Reservation_angular/src/app/service/waiter.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Router } from '@angular/router';
import { MenuItemReview } from "src/data/menuItemReview";
import { OrderItem } from 'src/data/orderItem';
import { SysImage } from 'src/data/sysImage';
import { Testimonial } from 'src/data/testimonial';
import { Waiter } from 'src/data/waiter';
import { environment } from 'src/environments/environment';
@Injectable({
  providedIn: 'root'
})
export class WaiterService {

  headerDict={
    'Content-Type':'application/json',
    'Accept':'application/json'
    }
 requestOptions={
      headers:new HttpHeaders(this.headerDict)
      }
 constructor(public http:HttpClient,private route:Router) { }
InsertWaiter(waiter:Waiter){

  
}
UpdateWaiter(waiter:Waiter){}
DeleteWaiterByWaiterId(WaiterId:number){}
GetAllWaiter(){}
GetAvailableWaiter(){
return this.http.get<any[]>(environment.apiUrl+'Waiter/GetAvailableWaiter',this.requestOptions);

}


}
    

File: Business Logic Layer(Backend)/Reservation/Reservation.Infra/Service/WaiterService.cs
using Reservation.Core.Data;
using Reservation.Core.Repository;
using Reservation.Core.Service;
using System;
using System.Collections.Generic;
using System.Text;

namespace Reservation.Infra.Service
{
    public class WaiterService: IWaiterService
    {
        private readonly IWaiterRepositroy _waiterRepositroy;
        public WaiterService(IWaiterRepositroy waiterRepositroy)
        {
            _waiterRepositroy = waiterRepositroy;
        }

        public bool DeleteWaiterByWaiterId(decimal WaiterId)
        {
            return _waiterRepositroy.DeleteWaiterByWaiterId(WaiterId);
        }

        public List<Waiter> GetAllWaiter()
        {
            return _waiterRepositroy.GetAllWaiter();
        }

        public List<Waiter> GetAvailableWaiter()
        {
            return _waiterRepositroy.GetAvailableWaiter();
        }

        public bool InsertWaiter(Waiter waiter)
        {
            return _waiterRepositroy.InsertWaiter(waiter);
        }

        public bool UpdateWaiter(Waiter waiter)
        {
            return _waiterRepositroy.UpdateWaiter(waiter);
        }
    }
}
